src/ble/CHIPBleServiceData.h

```diff
@@ -64,17 +66,31 @@ struct ChipBLEDeviceIdentificationInfo
 
     void SetProductId(uint16_t productId) { chip::Encoding::LittleEndian::Put16(DeviceProductId, productId); }
 
+    uint16_t GetAdvertisementVersion() const
+    {
+        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminatorAndAdvVersion) & kAdvertisementMask;
+    }
+
+    // Use only 4 bits to set advertisement version
+    void SetAdvertisementVersion(uint8_t advertisementVersion)
+    {
+        // Advertisement Version is 4 bit long from 12th to 15th
+        advertisementVersion &= kAdvertisementVersionMask;
+        advertisementVersion <<= 4u;
+        DeviceDiscriminatorAndAdvVersion[1] |= advertisementVersion;
+    }
+
     uint16_t GetDeviceDiscriminator() const
     {
-        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminator) & kDiscriminatorMask;
+        return chip::Encoding::LittleEndian::Get16(DeviceDiscriminatorAndAdvVersion) & kDiscriminatorMask;
     }
 
     void SetDeviceDiscriminator(uint16_t deviceDiscriminator)
     {
         // Discriminator is 12-bit long, so don't overwrite bits 12th through 15th
         deviceDiscriminator &= kDiscriminatorMask;
-        deviceDiscriminator |= static_cast<uint16_t>(DeviceDiscriminator[1] << 8u & ~kDiscriminatorMask);
-        chip::Encoding::LittleEndian::Put16(DeviceDiscriminator, deviceDiscriminator);
+        deviceDiscriminator |= static_cast<uint16_t>(DeviceDiscriminatorAndAdvVersion[1] << 8u & ~kDiscriminatorMask);
```

Comment:
ok , So you are storing the advertisementVersion from DeviceDiscriminatorAndAdvVersion[1] to not overwrite it while doing Put16 ? right also can you add a comment about that storage of advertisementVersion. seems you have addressed the priors comments

Reply:
added comment at declairation at DeviceDiscriminatorAndAdvVersion